todo-list/script.js

```diff
@@ -1,150 +1,231 @@
-            if(event.key === 'Enter' && todos[id].label == this.value) {
-                todoEdit.hidden = true;
-                renderTodos();
+    createStructureApp() {
+        // create wrapper app
+        const wrapperElementApp = this.createElement('section','','wrapper-todo-app');
+
+        // create Title
+        const h1 = this.createElement('h1');
+        h1.innerText = this.titleOfApp;
+        wrapperElementApp.appendChild(h1);
+
+        const wrapperInsertTask = this.createElement('section','', 'wrapper-insert');
+        const addTaskBtn = this.createElement('button', 'Add Task', 'add-task-btn');
+        const inputElement = this.createElement('input', '', 'input-text-task');
+        inputElement.setAttribute('type','text');
+        inputElement.setAttribute('placeholder','Enter New Task');
+        wrapperInsertTask.appendChild(addTaskBtn);
+        wrapperInsertTask.appendChild(inputElement);
+        wrapperElementApp.appendChild(wrapperInsertTask);
+
+        const wrapperListOfTasks = this.createElement('ul', '', 'wrapper-list-taks');
+        wrapperElementApp.appendChild(wrapperListOfTasks);
+
+        // get context this for use inside functions
+        const mainContext = this;
+        // function of event
+        function handlerAddTaskBtn(event) {
+            let value = '';
+            // console.log(this.tagName)
+            if(this.tagName == 'INPUT') {
+                value = this.value.trim();
+            } else if(this.tagName == 'BUTTON') {
+                value = this.closest('section').querySelector('.input-text-task').value;         
+            }
+            
+            const result = Boolean(mainContext.state.find(taskObj => taskObj.label == value));
+            if(result) {
+                alert('Your task is already have!!!')
+            }
+            if(value != '' && value != ' ' && !result) {
+                const newTask = new Task(value);
+                mainContext.state = [
+                    ...mainContext.state,
+                    newTask
+                ];
+                event.target.value = '';
+                mainContext.renderTodoTasks();
             }
+        };
+
+        // Add EventListeners
+        addTaskBtn.addEventListener('click', handlerAddTaskBtn);
+        inputElement.addEventListener('change', handlerAddTaskBtn);
+
+        this.elementInitApp.appendChild(wrapperElementApp);
+    }
 
-            // Clean the memory
-            todoEdit.removeEventListener('keypress', changeValueOfTodo);
+    renderTodoTasks() {
+        const fragment = document.createDocumentFragment();
+        const wrapperListOfTasks = this.elementInitApp.querySelector('.wrapper-todo-app .wrapper-list-taks');
+        wrapperListOfTasks.innerHTML = '';
+
+        this.state.forEach((taskObj, index) => {
+            const guid = this.genarateUID(5);
+
+            const li = this.createElement('li');
+            li.setAttribute('id', `${index}`);
```

Comment:
Why `${index}` ?

Reply:
i setted index of array to each li for find exact data